### PR TITLE
feat: post-liquidation storage

### PR DESCRIPTION
**Pull request by kjr217**
# Task: Post liquidation storage

## High Level Description

Store the details of a liquidation in a mapping struct after a liquidation, this can be queried until it is cleared by the vault owner.

### CODE REVIEW

File: contracts/core/Controller.sol [hunk 1]
```diff
@@ -447,6 +450,16 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
...
 
+    /**
+     * @notice clear a vaults liquidation details
+     * @param _owner account owner of the vault
+     * @param _vaultId vaultId to return balances for
+     */
+    function clearVaultLiquidationDetails(address _owner, uint256 _vaultId) external {
+        require(_owner == msg.sender);
+        delete vaultLiquidationDetails[_owner][_vaultId];
+    }
```
Comment (antoncoding) [inline contracts/core/Controller.sol:461]: how about just 
```
delete vaultLiquidateionDetails[msg.sender][_vaultId]
```

File: contracts/libs/MarginVault.sol [hunk 2]
```diff
@@ -47,6 +47,14 @@ library MarginVault {
         uint256[] collateralAmounts;
     }
 
+    // vaultLiquidationDetails is a struct of 3 variables that store the series address, short amount liquidated and collateral transferred for
+    // a given liquidation
+    struct VaultLiquidationDetails {
+        address series;
+        uint256 shortAmount;
+        uint256 collateralAmount;
```
Comment (antoncoding) [inline contracts/libs/MarginVault.sol:55]: I would say use uint128 here to save SSTORE. (but this gas saving doesn't matter too much on L2 tbh)

File: contracts/core/Controller.sol (hunk 1 above)
Comment (kjr217) [reply contracts/core/Controller.sol:461]: yeah good idea

File: contracts/libs/MarginVault.sol (hunk 2 above)
Comment (kjr217) [reply contracts/libs/MarginVault.sol:55]: yeah sure

Comment (kjr217) [reply contracts/libs/MarginVault.sol:55]: conversion is annoying thats all

**Review by antoncoding — APPROVED** [review]

**Pull request merged**